src/async-wrap.cc

```diff
@@ -177,6 +179,136 @@ static void PushBackDestroyId(Environment* env, double id) {
 }
 
 
+static bool PreCallbackExecution(AsyncWrap* wrap) {
+  AsyncHooks* async_hooks = wrap->env()->async_hooks();
+  if (wrap->env()->using_domains()) {
+    Local<Value> domain_v = wrap->object()->Get(wrap->env()->domain_string());
+    if (domain_v->IsObject()) {
+      Local<Object> domain = domain_v.As<Object>();
+      if (domain->Get(wrap->env()->disposed_string())->IsTrue())
+        return false;
+      Local<Value> enter_v = domain->Get(wrap->env()->enter_string());
+      if (enter_v->IsFunction()) {
+        if (enter_v.As<Function>()->Call(domain, 0, nullptr).IsEmpty()) {
+          FatalError("node::AsyncWrap::MakeCallback",
+                    "domain enter callback threw, please report this");
```

Comment:
@addaleax Fixed this here https://github.com/nodejs/node/pull/13000/commits/5717d75e97f4699b2c7eb706b10a129cc11da2f2.